packages/ontario-frontend-cli/resources/templates/ontario-create-app/tests/integration/packages.spec.njk

```diff
@@ -0,0 +1,89 @@
+{# Integration tests for projects created using Ontario Frontend #}/**
+ * Integration tests that check if the required and selected node modules were installed
+ */
+const { expect } = require("chai");
+const fs = require("fs");
+const path = require("path");
+
+/** Tests for required and optional packages
+ * 1. Checks if the required ontario-frontend core package is installed.
+ * 2. If selected, the optional ES-Lint or Prettier packages were installed with their associated packages and dependencies.
+ *
+ * @param {string} package - Name of the node package/module.
+ * @param {object} optionalPackages - Name of the optional packages like ES-lint and Prettier.
+ * @param {boolean} optionalPackages.enabled - Boolean value if the optional packages were selected or not through the CLI prompt.
+ * @param {array} optionalPackages.packages - A list of associated packages for a parent package.
+ * The boolean value should be captured from the CLI prompts.
+ * 
+ * @returns {boolean} Based on the test, the result will either be TRUE or FALSE (error).
+ */
+
+// A utility function to read package.json
+const readPackageJson = () => {
+  const packageJsonPath = path.resolve(__dirname, "../../..", "package.json");
+  return JSON.parse(fs.readFileSync(packageJsonPath, "utf8"));
+};
+
+// Checks if a package exists in package.json both in dependencies and dev dependencies
+const isPackageInstalled = (pkg) => {
+  const packageJson = readPackageJson();
+  return (
+    (packageJson.dependencies &&
+      packageJson.dependencies.hasOwnProperty(pkg)) ||
+    (packageJson.devDependencies &&
+      packageJson.devDependencies.hasOwnProperty(pkg))
+  );
+};
+
+describe("Integration: Installed packages", function () {
+  // Array of required packages – can add more to the list
+  const requiredPackages = ["@ongov/ontario-frontend"];
+
+  // Testing the required packages
+  requiredPackages.forEach((pkg) => {
+    it(`The core package "${pkg}" should be installed`, function () {
+      try {
+        if (!isPackageInstalled(pkg)) {
+          throw new Error();
+        }
+      } catch (e) {
+        expect.fail(`Failed: to detect the required package: "${pkg}"`);
+      }
+    });
+  });
+
+  // The optional packages and their associated packages
+  const optionalPackages = {
+    eslint: {
+      enabled: {{ addESLint }},
+      packages: [
+        "eslint",
+        "eslint-plugin-import",
+        "@ongov/eslint-config-ontario-frontend",
```

Comment:
I think it's worth importing the config objects for ESLint and Prettier and looking for them under the `packages` key. Keeps everything referencing a single source of truth.

You can find the config files under `ontario-frontend-cli/core/config/packages`. Within `index.js` they get combined and exported as `PACKAGES_CONFIG`.

Reply:
Realizing that since these test files get copied from the cli during project generation that this might be tricky. Perhaps a tests global variable file that makes a copy of the packageConfig `{...PACKAGE_CONFIG}` would retain it upon copy allowing it to be available for use? Worth exploring.

Reply:
I believe I would need to add scripts to make this data object accessible to the templates.
@matt-stjean, should I invest some time in that direction?

Reply:
@syed-ods I think it's worth investing some time into it. Give yourself a time box, giving yourself a time limit of how much you think you should spend on it. When your time is up, we can meet up and talk it through?